flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -353,6 +354,10 @@ public InputDependencyConstraint getInputDependencyConstraint() {
 		return getJobVertex().getInputDependencyConstraint();
 	}
 
+	public ExecutionVertexID getID() {
```

Comment:
It seems more proper to construct the `ExecutionVertexID` in the constructor only once, then all the other usages could get it directly. And `DefaultFailoverTopology` does not need to create `ExecutionVertexID` as well.

Reply:
Fixing `DefaultFailverTopology` can be a separate PR

Reply:
done